python/lsst/ts/observatory/control/maintel/mtcs.py

```diff
@@ -579,6 +579,101 @@ async def dome_el_in_position(self) -> str:
         await self._dome_el_in_position.wait()
         return "Dome elevation in position."
 
+    async def wait_for_dome_state(
+        self,
+        expected_states: set[MTDome.MotionState],
+        bad_states: set[MTDome.MotionState],
+        timeout: float,
+        check_in_position: bool = False,
+    ) -> None:
+        """Wait for a specific dome state.
+
+        Parameters
+        ----------
+        expected_states : set[MTDome.MotionState]
+            Valid states to transition into while un-parking.
+        bad_states : set[MTDome.MotionState]
+            States that are not allowed while un-parking and should raise an
+            error.
+        timeout : float
+            Maximum time to wait for the correct state.
+
+        Raises
+        ------
+        RuntimeError
+            If a bad state is encountered or the expected state is not reached
+             in time.
+        """
+        az_motion = await self.rem.mtdome.evt_azMotion.aget(timeout=self.long_timeout)
+        while az_motion.state not in expected_states:
+            az_motion = await self.rem.mtdome.evt_azMotion.next(
+                Flush=False, timeout=timeout
+            )
+
+            if az_motion.state in bad_states:
+                raise RuntimeError(
+                    f"Dome transitioned to an invalid state:" f" {az_motion.state}"
+                )
+
+            # If check_in_position is True, verify the dome is in the correct
+            # position.
+            if check_in_position and not az_motion.inPosition:
+                self.log.debug(
+                    f"Dome not yet in position. Current state: " f"{az_motion.state}."
+                )
+
+            self.log.debug(
+                f"Dome state: {az_motion.state}, inPosition: " f"{az_motion.inPosition}"
+            )
+
+            # If we reach this point, the dome has successfully reached an
+            # expected state.
+        if check_in_position and not az_motion.inPosition:
+            raise RuntimeError(
+                f"Dome reached the expected state "
+                f"{az_motion.state} but is not yet in position."
+            )
+
+    async def park_dome(self) -> None:
+        """Park the dome by moving it to the park azimuth."""
+        self.log.info(f"Parking dome at azimuth {self.dome_park_az}.")
```

Comment:
I think the MTDome has its own internal dome park position which could be different from `dome_park_az`. Consider making this only "Parking dome."